tape/src/main/java/com/squareup/tape/FileObjectQueue.java

```diff
@@ -75,6 +85,18 @@ public FileObjectQueue(File file, Converter<T> converter) throws IOException {
       queueFile.forEach(new QueueFile.ElementVisitor() {
         int count;
         @Override public boolean read(InputStream in, int length) throws IOException {
+          final int fileLength = queueFile.fileLength - QueueFile.HEADER_LENGTH;
+          if (length > fileLength || length > queueFile.maxElementSize) {
+            // Maybe fileLength is very small?
+            final int debugSize = Math.min(fileLength, Math.min(LARGE_ELEMENT_PREVIEW_SIZE, length));
+            byte[] debugBytes = new byte[debugSize];
+            final int readSize = in.read(debugBytes, 0, debugSize);
+            throw new CorruptionException("Possible corruption: Trying to read a Tape element "
+                + "claiming length: " + length + ", which is either a) larger than the current file "
+                + "length of: " + fileLength + ", or larger than the max Tape element size of: " +
+                queueFile.maxElementSize + ". The first " + readSize + " bytes of the element are: " +
+                new String(debugBytes, StandardCharsets.UTF_8));
```

Comment:
I don' think a string works well here as we'll likely get loss due to encoding charsets. base64 would ensure no loss

Reply:
Good call.

Reply:
Set to Base64 encoding.